src/vs/workbench/contrib/terminal/browser/links/terminalLinkManager.ts

```diff
@@ -229,6 +229,9 @@ export class TerminalLinkManager extends DisposableStore {
 			}
 		}
 
+		// encode the uri so TerminalLink get rendered as anchor element https://github.com/microsoft/vscode/issues/108301
+		uri = encodeURI(uri);
```

Comment:
Suggestion: while changing this section how about moving the undefined check out of the template and naming the variable so not to give the impression the function attribute to being mutated.
```
const encodedUri = encodeURI(uri) || 'undefined';
return new MarkdownString(`[${label || nls.localize('followLink', "Follow Link")}](${encodedUri}) (${clickLabel})`, true);
	
```

Reply:
Good idea. I'll update the code